uber/model_checks.py

```diff
@@ -264,24 +258,41 @@ def dealer_needs_group(attendee):
 
 @validation.Attendee
 def dupe_badge_num(attendee):
-    if c.NUMBERED_BADGES and attendee.badge_num != 0 and attendee.session.query(Attendee)\
-            .filter(Attendee.id != attendee.id)\
-            .filter_by(badge_type=attendee.badge_type, badge_num=attendee.badge_num).count():
-        return 'Another attendee already exists with that badge number!'
+    if (attendee.badge_num != attendee.orig_value_of('badge_num') or attendee.is_new)\
+            and c.NUMBERED_BADGES and attendee.badge_num and not c.SHIFT_CUSTOM_BADGES:
+        with Session() as session:
+            existing = session.query(Attendee)\
+                .filter_by(badge_type=attendee.badge_type, badge_num=attendee.badge_num)
+            if existing.count():
+                return 'That badge number already belongs to {!r}'.format(existing.first().full_name)
 
 
 @validation.Attendee
 def invalid_badge_num(attendee):
-    if c.NUMBERED_BADGES:
+    if c.NUMBERED_BADGES and attendee.badge_num:
         try:
```

Comment:
This try/except should theoretically never run because we now simply set badge_num to None if it's not an int, but I haven't gone through the code and made sure that will happen in every case. Consider that a TODO.